packages/core/src/bind/connectFactoryObservable.ts

```diff
@@ -53,17 +56,21 @@ export default function connectFactoryObservable<A extends [], O>(
       },
     )
 
-    const reactObservable$ = reactEnhancer(sharedObservable$)
-
-    const publicShared$: Observable<O> = defer(() => {
+    const publicShared$ = new Observable<O>((subscriber) => {
       const inCache = cache.get(keys)
-      if (inCache) {
-        return inCache[0] === publicShared$ ? sharedObservable$ : inCache[0]
-      }
-      return getSharedObservables$(input)[0]
-    })
+      const source$ = inCache
+        ? inCache[0] === publicShared$
+          ? sharedObservable$
+          : inCache[0]
+        : getSharedObservables$(input)[0]
+
+      publicShared$.getValue = source$.getValue
+
+      return source$.subscribe(subscriber)
+    }) as BehaviorObservable<O>
```

Comment:
Something calling `.getValue` before no subscription happens will throw an error. Would it be safer to pre-initialize getValue to ~~`throw EMPTY_VALUE`~~ `sharedObservable$.getValue`?

Reply:
> Something calling .getValue before no subscription happens will throw an error.

Yep, that is intentional.

> Would it be safer to pre-initialize getValue to throw EMPTY_VALUE sharedObservable$.getValue?

With the current implementation as long as it throws, it doesn't matter what it throws. It would be a lot more clear, for sure, but it wouldn't make a difference.